types/koa-multer/index.d.ts

```diff
@@ -0,0 +1,114 @@
+            fields?: number;
+            /** For multipart forms, the max file size (in bytes)(Default: Infinity) */
+            fileSize?: number;
+            /** For multipart forms, the max number of file fields (Default: Infinity) */
+            files?: number;
+            /** For multipart forms, the max number of parts (fields + files)(Default: Infinity) */
+            parts?: number;
+            /** For multipart forms, the max number of header key=> value pairs to parse Default: 2000(same as node's http). */
+            headerPairs?: number;
+            /** Keep the full path of files instead of just the base name (Default: false) */
+            preservePath?: boolean;
+        };
+        /** A function to control which files to upload and which to skip. */
+        fileFilter?(req: IncomingMessage, file: File, callback: (error: Error | null, acceptFile: boolean) => void): void;
+    }
+
+    interface StorageEngine {
+        _handleFile(req: IncomingMessage, file: File, callback: (error?: any, info?: File) => void): void;
+        _removeFile(req: IncomingMessage, file: File, callback: (error: Error) => void): void;
+    }
+
+    interface DiskStorageOptions {
+        /** A function used to determine within which folder the uploaded files should be stored. Defaults to the system's default temporary directory. */
+        destination?: string | ((req: IncomingMessage, file: File, callback: (error: Error | null, destination: string) => void) => void);
+        /** A function used to determine what the file should be named inside the folder. Defaults to a random name with no file extension. */
+        filename?(req: IncomingMessage, file: File, callback: (error: Error | null, filename: string) => void): void;
+    }
+
+    interface Instance {
+        /** In case you need to handle a text-only multipart form, you can use any of the multer methods (.single(), .array(), fields()), req.body contains the text fields */
+        /** Accept a single file with the name fieldName. The single file will be stored in req.file. */
+        single(fieldName?: string): Koa.Middleware;
+        /** Accept an array of files, all with the name fieldName. Optionally error out if more than maxCount files are uploaded. The array of files will be stored in req.files. */
+        array(fieldName: string, maxCount?: number): Koa.Middleware;
+        /** Accept a mix of files, specified by fields. An object with arrays of files will be stored in req.files. */
+        fields(fields: Field[]): Koa.Middleware;
+        /** Accepts all files that comes over the wire. An array of files will be stored in req.files. */
+        any(): Koa.Middleware;
+    }
+}
+
+interface Multer {
+    (options?: multer.Options): multer.Instance;
+
+    /* The disk storage engine gives you full control on storing files to disk. */
+    diskStorage(options: multer.DiskStorageOptions): multer.StorageEngine;
```

Comment:
You could put these as functions inside the namespace instead, and `declare function multer(options?: multer.Options): multer.Instance` instead of the `declare const`.